app/recordtransfer/utils.py

```diff
@@ -354,3 +356,18 @@ def accept_session(filename: str, filesize: Union[str, int], session: "UploadSes
 
     # All checks succeded
     return {"accepted": True}
+
+
+def get_js_translation_version() -> str:
+    """Return a hash of the latest modification time of all djangojs.mo files in the locale
+    directory. This changes whenever compiled JS translations are updated.
+    """
+    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    locale_dir = os.path.join(base_dir, "locale")
+    mo_files = glob.glob(os.path.join(locale_dir, "*/LC_MESSAGES/djangojs.mo"))
+    latest_mtime = 0
+    for mo_file in mo_files:
+        mtime = os.path.getmtime(mo_file)
+        if mtime > latest_mtime:
+            latest_mtime = mtime
+    return hashlib.md5(str(latest_mtime).encode()).hexdigest()
```

Comment:
There are a few simplifications we can make to this function:

- Use `max` instead of manually tracking the latest time.
- Use `LOCALE_PATHS` instead of finding them yourself.
- Use `pathlib.Path` instead of glob. It has a very nice [rglob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob) function
- Just use the time instead of hashing the time. The time is unique enough, it doesn't also need to be hashed.

```suggestion
def get_js_translation_version() -> str:
    """Return the latest modification time of all djangojs.mo files in the locale directory.

    This changes whenever compiled JS translations are updated.
    """
    return str(
        max(
            [
                item.stat().st_mtime
                for locale_dir in settings.LOCALE_PATHS
                for item in Path(locale_dir).rglob("djangojs.mo")
            ]
        )
    )
```

Reply:
Beautiful - so much cleaner.